filters/diag/diag.go

```diff
@@ -0,0 +1,344 @@
+/*
+Package diag provides a set of network throttling filters for diagnostic purpose.
+
+The filters enable adding artificial latency, limiting bandwidth or chunking responses with custom chunk size
+and delay. This throttling can be applied to the proxy responses or to the outgoing backend requests. An
+additional filter, randomContent, can be used to generate response with random text of specified length.
+*/
+package diag
+
+import (
+	"io"
+	"math/rand"
+	"net/http"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/zalando/skipper/filters"
+	"github.com/zalando/skipper/filters/serve"
+)
+
+const defaultChunkSize = 512
+
+const (
+	RandomName           = "randomContent"
+	LatencyName          = "latency"
+	ChunksName           = "chunks"
+	BandwidthName        = "bandwidth"
+	BackendLatencyName   = "backendLatency"
+	BackendBandwidthName = "backendBandwidth"
+	BackendChunksName    = "backendChunks"
+)
+
+type throttleType int
+
+const (
+	latency throttleType = iota
+	bandwidth
+	chunks
+	backendLatency
+	backendBandwidth
+	backendChunks
+)
+
+type random struct {
+	len int
+}
+
+type throttle struct {
+	typ       throttleType
+	chunkSize int
+	delay     time.Duration
+}
+
+var randomChars = []byte("ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789")
+
+func kbps2bpms(kbps float64) float64 {
+	return kbps * 1024 / 1000
+}
+
+// NewRandom creates a filter specification whose filter instances can be used
+// to respond to requests with random text of specified length. It expects the
+// the byte length of the random response to be generated as an argument.
+// Eskip example:
+//
+// 	* -> randomContent(2048) -> <shunt>;
+//
+func NewRandom() filters.Spec { return &random{} }
+
+// NewLatency creates a filter specification whose filter instances can be used
+// to add additional latency to responses. It expects the latency in milliseconds
+// as an argument. It always adds this value in addition to the natural latency,
+// and does not do any adjustments. Eskip example:
+//
+// 	* -> latency(120) -> "https://www.example.org";
+//
+func NewLatency() filters.Spec { return &throttle{typ: latency} }
```

Comment:
I would've preferred that all the latency arguments to be strings and accept valid ParseDuration values. We could still accept simple numbers and default that to ms. It's a matter of taste but I'd like to see arguments such as "1s", "500µs", etc.
One thing to filter out, just like right now, would be negative latencies.